### ISSUE TITLE
Reduce Not loading in Julia 0.6.2

### ISSUE THREAD

**Issue by mattcbro** [chakravala/Reduce.jl#13]
I'm getting the following error:

julia> using Reduce
INFO: Precompiling module Reduce.
ERROR: InitError: MethodError: no method matching contains(::String, ::Regex)
Closest candidates are:
  contains(::Function, ::Any, ::Any) at reduce.jl:664
  contains(::AbstractString, ::AbstractString) at strings/search.jl:378
Stacktrace:
 [1] ReduceCheck at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:60 [inlined]
 [2] macro expansion at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:209 [inlined]
 [3] anonymous at ./<missing>:?
 [4] Load() at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:224
 [5] __init__() at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:191
 [6] _include_from_serialized(::String) at ./loading.jl:157
 [7] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [8] _require(::Symbol) at ./loading.jl:498
 [9] require(::Symbol) at ./loading.jl:405
during initialization of module Reduce

**Comment by mattcbro** [comment]
I think I had not yet installed reduce.  However it still won't load.  My new error is:
ERROR: InitError: write: broken pipe (EPIPE)
Stacktrace:
 [1] try_yieldto(::Base.##296#297{Task}, ::Task) at ./event.jl:189
 [2] wait() at ./event.jl:234
 [3] uv_write(::Base.PipeEndpoint, ::Ptr{UInt8}, ::UInt64) at ./stream.jl:811
 [4] unsafe_write(::Base.PipeEndpoint, ::Ptr{UInt8}, ::UInt64) at ./stream.jl:832
 [5] write(::Pipe, ::String) at ./strings/string.jl:71
 [6] macro expansion at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:200 [inlined]
 [7] anonymous at ./<missing>:?
 [8] Load() at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:224
 [9] __init__() at /home/matt/programs/juliapro/JuliaPro-0.6.2.1/JuliaPro/pkgs-0.6.2.1/v0.6/Reduce/src/Reduce.jl:191
 [10] _include_from_serialized(::String) at ./loading.jl:157
 [11] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [12] _require_search_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:236
 [13] _require(::Symbol) at ./loading.jl:441
 [14] require(::Symbol) at ./loading.jl:405
during initialization of module Reduce

**Comment by chakravala** [comment]
What OS are you using? You can see on the travis ci and the appveyor tests that Reduce.jl works fine and is tested with 0.6.2 on Linux, OSX, and Windows. However, it has not been tested on JuliaPro by me, and it looks like you are a JuliaPro user. Unfortunately, I am not able to reproduce your issue, so I cant know where the issue comes from. Have you tried the standalone version of Julia 0.6.2?

**Comment by mattcbro** [comment]
I was using Juliapro 0.6.2.1 and tried to upgrade to 0.6.2.2.  My machine is linux mint 18.3.  I am wondering what are the installation requirements for reduce prior to installation of this package?

**Comment by chriselrod** [comment]
`Pkg.add("Reduce")` is all it should take.

I tried JuliaPro around a year ago, but found it too awkward to upgrade and manage packages compared to a normal install.

**Comment by chakravala** [comment]
Yes, as pointed out by chris, there are no special requirements, aside from Julia 0.6, you can also use `Pkg.build("Reduce")` to retrigger the build script, but if the binaries are already downloaded and up to date nothing will be changed. The binaries for the upstream reduce are portable and dont have any dependencies, the only thing needed is Julia 0.6. It's also possible to compile the upstream reduce from source or otherwise install it on your /usr/bin path, and it will automatically be detected and used. But this looks like some issue that is specific to JuliaPro.

I am not familiar with JuliaPro, and am quite busy through the next week, so I cant guarantee that this will be immediately resolved, but I encourage you or chris (being more familiar with JuliaPro), to try to figure out the source of the issue or provide more detailed info about reproducing the issue.
